src/Migrations.php

```diff
@@ -0,0 +1,280 @@
+ */
+namespace Migrations;
+
+use Phinx\Config\ConfigInterface;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Output\NullOutput;
+
+/**
+ * The Migrations class is responsible for handling migrations command
+ * within an none-shell application.
+ */
+class Migrations
+{
+
+    use ConfigurationTrait;
+
+    /**
+     * The OutputInterface.
+     * Should be a \Symfony\Component\Console\Output\NullOutput instance
+     *
+     * @var \Symfony\Component\Console\Output\OutputInterface
+     */
+    protected $output;
+
+    /**
+     * CakeManager instance
+     *
+     * @var \Migrations\CakeManager
+     */
+    protected $manager;
+
+    /**
+     * The last error caught.
+     *
+     * @var string
+     */
+    protected $lastError;
+
+    /**
+     * Default options to use
+     *
+     * @var array
+     */
+    protected $default = [];
+
+    /**
+     * Constructor
+     * @param array $default Default option to be used when calling a method.
+     * Available options are :
+     * - `connection` The datasource connection to use
+     * - `source` The folder where migrations are in
+     * - `plugin` The plugin containing the migrations
+     */
+    public function __construct(array $default = [])
+    {
+        $this->output = new NullOutput();
+
+        if (!empty($default)) {
+            $this->default = $default;
+        }
+    }
+
+    /**
+     * Returns the status of each migrations based on the options passed
+     *
+     * @param array $options Options to pass to the command
+     * Available options are :
+     *
+     * - `format` Format to output the response. Can be 'json'
+     * - `connection` The datasource connection to use
+     * - `source` The folder where migrations are in
+     * - `plugin` The plugin containing the migrations
+     *
+     * @return array The migrations list and their statuses
+     */
+    public function status($options = [])
+    {
+        $input = $this->getInput('Status', [], $options);
+        $params = ['default', $input->getOption('format')];
+
+        return $this->run('printStatus', $params, $input);
+    }
+
+    /**
+     * Migrates available migrations
+     *
+     * @param array $options Options to pass to the command
+     * Available options are :
+     *
+     * - `target` The version number to migrate to. If not provided, will migrate
+     * everything it can
+     * - `connection` The datasource connection to use
+     * - `source` The folder where migrations are in
+     * - `plugin` The plugin containing the migrations
+     *
+     * @return bool Success
+     */
+    public function migrate($options = [])
+    {
+        $input = $this->getInput('Migrate', [], $options);
+        $params = ['default', $input->getOption('target')];
+
+        try {
+            $this->run('migrate', $params, $input);
+            return true;
+        } catch (\Exception $e) {
```

Comment:
Would it be too bad to just let the exception bubble up? I think it is better than jus returning false